### PR TITLE
Feature/ export order data as excel

### PR DESCRIPTION
**Pull request by pasindu-clouda**
#### What does this PR do? \*

- add download feature to get summary of order details

workspace: [https://quickorderdev--sbdsefuat.myvtex.com/](https://quickorderdev--sbdsefuat.myvtex.com/quickorder?__bindingAddress=catalog.sef.ao.sbd-nonprod.com/en-US/catalog-global)

### CODE REVIEW

File: react/TextAreaBlock.tsx [hunk 1]
```diff
@@ -61,6 +66,15 @@ const TextAreaBlock: FunctionComponent<
   const [refidLoading, setRefIdLoading] = useState<any>()
   const [loading, setLoading] = useState<boolean>(false)
   const [isModalOpen, setIsModelOpen] = useState<boolean>(false)
+  const [base64Image, setBase64Image] = useState('')
+  const [exceldownloading, setExcelDownloading] = useState<boolean>(false)
```
Comment (sahanljc) [inline react/TextAreaBlock.tsx]: Use `excelDownloading`.

File: react/TextAreaBlock.tsx [hunk 2]
```diff
@@ -61,6 +66,15 @@ const TextAreaBlock: FunctionComponent<
...
+  const [isMTO, setIsMTO] = useState<boolean>(false)
```
Comment (sahanljc) [inline react/TextAreaBlock.tsx]: Use `isMto` and `setIsMto`.

File: react/TextAreaBlock.tsx [hunk 3]
```diff
@@ -421,17 +678,37 @@ const TextAreaBlock: FunctionComponent<
...
-                isLoading={mutationLoading}
-                onClick={() => {
-                  addToCartCopyNPaste()
-                }}
-                disabled={!showAddToCart}
-              >
-                <FormattedMessage id="store/quickorder.addToCart" />
-              </Button>
+              <div className="flex justify-between">
+                <div style={{ marginRight: '12px' }}>
```
Comment (sahanljc) [inline react/TextAreaBlock.tsx:680]: Get rid of hard-coded margin if possible.

File: react/UploadBlock.tsx [hunk 4]
```diff
@@ -146,6 +187,9 @@ const UploadBlock: FunctionComponent<
         textAreaValue: GetText(items),
       })
 
+      console.log('adding items to context: upload')
```
Comment (sahanljc) [inline react/UploadBlock.tsx]: Remove log if not required.

File: react/UploadBlock.tsx [hunk 5]
```diff
@@ -262,6 +306,80 @@ const UploadBlock: FunctionComponent<
     doFile(files)
   }
 
+  // const processWorkbook = async (file: File) => {
```
Comment (sahanljc) [inline react/UploadBlock.tsx]: Remove section if not required.

File: react/UploadBlock.tsx [hunk 6]
```diff
@@ -425,6 +543,9 @@ const UploadBlock: FunctionComponent<
     return <p>Loading sold to..</p>
   }
 
+  console.log('reviewItems: upload', reviewItems)
```
Comment (sahanljc) [inline react/UploadBlock.tsx]: Remove logging if not required.

File: react/components/ReviewBlock.css [hunk 7]
```diff
@@ -46,10 +46,19 @@
 
 .quickOrderTableRowWithIssues {
   background-color: #ffd6d6;
+  height: 150px;
```
Comment (sahanljc) [inline react/components/ReviewBlock.css:49]: Be careful about hard-coded styles.

File: react/components/ReviewBlock.tsx [hunk 8]
```diff
@@ -244,6 +256,9 @@ const ReviewBlock: FunctionComponent<WrappedComponentProps & any> = ({
   onRefidLoading,
   intl,
 }: any) => {
+  console.log('props: onReviewItems', onReviewItems)
```
Comment (sahanljc) [inline react/components/ReviewBlock.tsx]: Remove logging.

File: react/components/ReviewBlock.tsx [hunk 9]
```diff
@@ -458,6 +473,8 @@ const ReviewBlock: FunctionComponent<WrappedComponentProps & any> = ({
         const sellers = item.sku ? mappedRefId[item.sku]?.sellers : '1'
         const itm = getItemFromQuery(item)
 
+        console.log('itm', itm)
```
Comment (sahanljc) [inline react/components/ReviewBlock.tsx]: Remove logging if not required.

File: react/components/StockAvailability.tsx [hunk 10]
```diff
@@ -81,6 +85,24 @@ const StockAvailability = ({
...
+      data?.getStockAvailability?.qtyAvailable ?? '0',
+      10
+    )
+
+    handleExtractData(itemNumber, stockAvailability, 'stockAvailability')
+  }
+
+  useEffect(() => {
+    if (!loading) {
+      console.log('add stock availability to context')
```
Comment (sahanljc) [inline react/components/StockAvailability.tsx]: Remove logging.

File: react/utils/buildURL.ts [hunk 11]
```diff
@@ -0,0 +1,11 @@
+export const buildURL = (
```
Comment (sahanljc) [inline react/utils/buildURL.ts]: Use `buildUrl`.

**Review by sahanljc — CHANGES_REQUESTED** [review]

File: react/components/ReviewBlock.css (hunk 7 above)
Comment (pasindu-clouda) [reply react/components/ReviewBlock.css:49]: I have discussed it with Nadeera ayya

File: manifest.json [hunk 12]
```diff
@@ -101,6 +101,13 @@
         "path": "/api/dataentities/*"
       }
     },
+    {
```
Comment (sahanljc) [inline manifest.json:104]: Is this required?

**Review by sahanljc — CHANGES_REQUESTED** [review]

File: manifest.json (hunk 12 above)
Comment (pasindu-clouda) [reply manifest.json:104]: These changes were made by Harsha ayya and its related to the styling exported excel sheet

**Review by sahanljc — APPROVED** [review]

**Pull request merged**